src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EmbeddedLanguages/Common/EmbeddedSyntaxToken.cs

```diff
@@ -93,7 +93,7 @@ public string ToFullString()
         /// <param name="trailing">If false, trailing trivia will not be added</param>
         public void WriteTo(StringBuilder sb, bool leading, bool trailing)
         {
-            if (leading)
+            if (leading && !LeadingTrivia.IsDefault)
```

Comment:
Without this check, optional children would throw on `ToFullString` because everything is the default for an `EmbeddedSyntaxToken`

Reply:
interesting... LeadingTrivia should probably be non-default... but that can be a change in another PR.